### PR TITLE
fix(time synchronization check): Added full service print

### PR DESCRIPTION
**Pull request by ShlomiBalalis**
In the artifact test, we have a test that checks that one (and only one)
time synchronization service active. However, if the check fails and the number
of time sync services does not meet our expectations (either zero or more than one)
the error message is not enough to understand the issue. So, I added a log message
of all of the services in the node under test that is printed when the check fails,
so error will be clearer.

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by fgelcer** [comment]
would this be related to https://github.com/scylladb/scylla/issues/10608 in any sort?

**Review by yarongilor — COMMENTED** [review]
1. you may consider an alternative suggestion for that - just print all time-sync services in advance, and save the `try-except` block like:
```
full_list = self.node.remoter.run('systemctl list-units --full' | egrep -i 'timesync|ntp|chrony').stdout.strip()
                    self.log.debug("Installed timesync services: %s", full_list)
```
2. It appears that the last check might be incorrect following https://github.com/scylladb/scylla/issues/10608#issuecomment-1133935711

**Comment by rayakurl** [comment]
> would this be related to [scylladb/scylla#10608](https://github.com/scylladb/scylla/issues/10608) in any sort?

Yes

File: artifacts_test.py [hunk 1]
```diff
@@ -336,15 +336,21 @@ def test_scylla_service(self):
...
+                        assert not is_chrony_service_installed, \
+                            "systemd-timesyncd is already installed, yet scylla installed chrony service on top of it"
+                    else:
+                        assert is_ntp_service_installed or is_chrony_service_installed, \
+                            "systemd-timesyncd is not installed on the instance, yet Scylla did not install ntp or " \
+                            "chrony services as a replacement"
+                except AssertionError:
+                    full_list = self.node.remoter.run('systemctl list-units --full').stdout.strip()
+                    self.log.warning("Seems that there was an issue with ntp check. Here's the full list of services "
+                                     "active on the node: {}".format(full_list))
```
Comment (enaydanov) [inline artifacts_test.py:352]: Please, use lazy-logging.

**Review by enaydanov — CHANGES_REQUESTED** [review]

**Review by rayakurl — APPROVED** [review]
Approved

**Review by fruch — APPROVED** [review]
LGTM

**Comment by rayakurl** [comment]
@fruch , @enaydanov - can one of you please backport? thanks

**Comment by yarongilor** [comment]
@roydahan , this PR not following the suggested solution. i think it should be reverted.

**Comment by fgelcer** [comment]
> @roydahan , this PR not following the suggested solution. i think it should be reverted.

Avi suggested that NTP (or chrony) must be installed by the user, in advance... what was done here is just a more clear logging of the error, but the test will continue to fail.
this PR doesn't need to be reverted, but it won't "fix" the problem we originally had... so, please, @yarongilor send a PR, where you will install NTP or Chrony, before we start the offline installation.

**Comment by vponomaryov** [comment]
@fgelcer @fruch I see that the commit/PR name is confusing here. It must have been "improvement", not "fix".

**Comment by fgelcer** [comment]
> @fgelcer @fruch I see that the commit/PR name is confusing here. It must have been "improvement", not "fix".

i think you are right, but it is fixing the logging of the checks we run for the time sync packages... both are right, but yes, `improvement` would be more accurate.

**Comment by fgelcer** [comment]
> @fgelcer , @yarongilor - please sync with @ShlomiBalalis regarding this task. @ShlomiBalalis re-worked the code several times with several solutions. Yaron, Avi and Takuya asked for apposite things. Shlom followed Avi's request.

not really, as Avi said the package must be installed by user in advance, and we are not installing anything new for now... Shlomo just improved the logging, and did no change to these jobs AFAICT

**Comment by roydahan** [comment]
The question is if the user didn’t install any NTP package, will the installation pass?
If so, the test should pass and it’s ok to leave it as is.

**Pull request merged**